akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressionTable.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.remote.artery.compress
+
+/**
+ * Handles counting and detecting of heavy-hitters and compressing them via a table lookup.
+ * Mutable and not thread-safe.
+ *
+ * Compression flow goes like:
+ * [1]  on each message we add the actor path here
```

Comment:
Will we have one compression table per stream or one for all streams of the destination node?
If it is the latter it might be problematic with thread safety, since this will be updated by multiple threads (stream stages).
How costly is `addAndEstimateCount`? Would it be worth doing the updates async in the background (e.g. by an actor), but we also want to keep allocations low?
We could also use sampling if too costly to do on each message.